### DISCUSSION TITLE
How to increase the running speed of the MPC

### DISCUSSION THREAD

**Discussion by zmdntnu** [do-mpc/do-mpc#261] [General]
Hi, 

Do you have any idea how we can increase the running speed of the MPC? In my case, it will take about 1 hour to finish the simulation with a total iteration of 204, but this is too slow for the application. Could you please provide me with some suggestions regarding this issue? 

Thank you in advance.

Best wishes

zmdntnu

**Comment by 4flixt** [comment]
Dear @zmdntnu,

if you don't want to do major changes the only recommendation I can give is to choose a different linear solver for IPOPT. Here is some information on that: 

https://www.do-mpc.com/en/latest/installation.html#hsl-linear-solver-for-ipopt
